client/landing/stepper/declarative-flow/internals/hooks/use-flow-navigation/index.tsx

```diff
@@ -110,8 +111,14 @@ export const useFlowNavigation = ( flow: Flow ): FlowNavigation => {
 				...extraData,
 			} );
 
+			// For HOSTED_SITE_MIGRATION_FLOW, we must use the flow.variantSlug instead of flow.name
+			// to prevent navigation errors.
+			const exceptionFlowsToUseVariantSlug = [ HOSTED_SITE_MIGRATION_FLOW ];
```

Comment:
Trying to keep things as contained as possible as this is an escalation: p1736747265983289-slack-C02FMH4G8